assignmentScoring/assignmentScoring.ts

```diff
@@ -0,0 +1,97 @@
+enum Role {
```

Comment:
FYI I'm still planning to improve this code as part of thinking through assignment prioritization, sharing it now so that Zak can give a rough estimate of dev hours to implement something like this.

I do think it's far enough along that feedback about the proposed scoring rules would be helpful

I'd say don't worry too much about the code quality, we're not planning to actually use this code :)